### PR TITLE
Added retry payment option after payment is failed

### PR DESCRIPTION
**Pull request by harshvitra**
Only works in Staging. Wait for Backend release before merging to Production.

### CODE REVIEW

**Comment by sagararyal** [comment]
400 Error:

https://donate-with-planet-h6lic5rwr-planetapp.vercel.app/?payment_intent=pi_3JuI0KD2OpW2f42N00t9kLh7&payment_intent_client_secret=pi_3JuI0KD2OpW2f42N00t9kLh7_secret_aQvqaYpY6CrU7i21nQI580NSW&source_type=sofort&method=Sofort&redirect_status=failed&context=don_CeQTmV0fzc5PgwzNABsed2G5&tenant=ten_I9TW3ncG&step=payment

Amount is being passed null during retry
```
{"project":"proj_DbGBDHbePvTCUgndUcvrJodr","amount":null,"currency":"EUR","donor":{"firstname":"Peter","lastname":"Planter","email":"email@email.me","address":"Am Bahnhof 1","city":"Uffing Am Staffelsee","zipCode":"82449","country":"DE","companyname":null},"frequency":null,"quantity":86.3,"taxDeductionCountry":"DE"}
```

**Comment by sagararyal** [comment]
```
{"purpose":"trees","project":"proj_eNxsxTaYnt48i28ZVfFYAqbm","amount":20,"currency":"EUR","donor":{"firstname":"Peter","lastname":"Planter","email":"email@test.de","address":"123","city":"Munich","zipCode":"80539","country":"DE"},"frequency":null,"quantity":50,"taxDeductionCountry":"DE"}
```


```
{"project":"proj_eNxsxTaYnt48i28ZVfFYAqbm","amount":null,"currency":"EUR","donor":{"firstname":"Sagar","Peter":"Planter","email":"email@test.de","address":"123","city":"Munich","zipCode":"80539","country":"DE","companyname":null},"frequency":null,"quantity":20,"taxDeductionCountry":"DE"}
```
**missing:**
"purpose":"trees"

**incorrect:**
amount":null,
companyname":null (shouldn't be added, if null)
"quantity":20,"  incorrect amount ≠ quantity.

**Comment by sagararyal** [comment]
Screen Shot 2021-11-10 at 3 58 27 PM

- [ ]  in case of any error like this, = 4xx or 5xx send user to failed payment page.

File: src/Donations/Micros/PaymentStatus/FailedDonation.tsx [hunk 1]
```diff
@@ -4,37 +4,118 @@ import PaymentFailedIllustration from "../../../../public/assets/icons/donation/
...
+    }
+  }
+  async function getDonation() {
+    setdonationStep(3);
+    router.push({
+      query: { ...router.query, step: "payment" },
+    });
+    setIsTaxDeductible(donation.taxDeductionCountry);
+    setprojectDetails(donation.project);
+    setquantity(donation.amount);
```
[the remark this answers is not in the document]
Comment (sagararyal) [reply src/Donations/Micros/PaymentStatus/FailedDonation.tsx:66]: Add a line
`setamount(donation.amount);`

[the remark this answers is not in the document]
Comment (tejassonar) [reply src/Donations/Micros/PaymentStatus/FailedDonation.tsx:66]: Yes..!!

**Comment by tejassonar** [comment]
> Screen Shot 2021-11-10 at 3 58 27 PM
> 
> * [ ]   in case of any error like this, = 4xx or 5xx send user to failed payment page.

This thing is already fixed in recurrency PR.

**Review by ankitecd — COMMENTED** [review]
Works properly for me.

**Pull request merged**